src/domain/_common/components/Avatar/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { ReactComponent as AvatarIcon } from '@/assets/avatar-icon.svg';
+import generateRandomColor from '@/utils/generateRandomColor';
+
+interface AvatarProfileProps {
+  imageUrl?: string;
+  isSmall?: boolean;
+  userName?: string;
+}
+
+interface AvatarProps extends AvatarProfileProps {
+  type: 'icon' | 'column' | 'row';
+}
+
+interface AvatarListProps extends AvatarProfileProps {
+  userName: string;
+}
+
+export default function Avatar({ type, imageUrl, userName = '', isSmall = false }: AvatarProps) {
+  switch (type) {
+    case 'icon':
+      return <AvatarProfile imageUrl={imageUrl} isSmall={isSmall} />;
+    case 'column':
+      return <AvatarProfile.listColumn imageUrl={imageUrl} userName={userName} isSmall={isSmall} />;
+    case 'row':
+      return <AvatarProfile.listRow imageUrl={imageUrl} userName={userName} isSmall={isSmall} />;
+    default:
+      return null;
+  }
+}
```

Comment:
```suggestion
export default function Avatar({ type, imageUrl, userName = '', isSmall = false }: AvatarProps) {
  return {
    icon: <AvatarProfile imageUrl={imageUrl} isSmall={isSmall} />,
    column: <AvatarProfile.listColumn imageUrl={imageUrl} userName={userName} isSmall={isSmall} />,
    row: <AvatarProfile.listRow imageUrl={imageUrl} userName={userName} isSmall={isSmall} />,
  }[type];
}
```

이렇게 object 형식으로 받는건 어때요?